### PR TITLE
http client factory - autorest

### PR DESCRIPTION
**Pull request by VaclavK**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -126,3 +126,40 @@ internal class SecurityResponseTelemetryInitializer : Eshopworld.Web.Telemetry.R
...
+
+For further information re http client factory, see https://docs.microsoft.com/en-us/dotnet/standard/microservices-architecture/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests
+
+The helper returns http client builder chain - just like Microsoft extensions - and so the client can be further configured (e.g. with Polly).
+
+Usage
+
+``` c#
+services
+	.AddAutoRestTypedClient<ISwaggerPetstore, SwaggerPetstore>()
```
Comment (dguerin) [inline README.md]: Do we need to specify type in the name of the method? seems to be implicit since you specify the type.

Comment (VaclavK) [reply README.md]: I am not sure what you mean here.. will ping you on slack

Comment (dguerin) [reply README.md]: AddAutoRestTypedClient vs AddAutoRestClient

**Comment by VaclavK** [comment]
> 
> 
> Looks good, just one small query on name of method. Would certainly be worth the effort to introduce an example of this into the template as well.

would that not duplicate autorest and other doco? the snippet in readme is kinda ALL you do to plug it in anyway

File: README.md (hunk 1 above)
Comment (VaclavK) [reply README.md]: yeah .. as explained on slack - this was to tie in to http client /MS terminology - typed vs named clients e.g.

**Comment by dguerin** [comment]
> > Looks good, just one small query on name of method. Would certainly be worth the effort to introduce an example of this into the template as well.
> 
> would that not duplicate autorest and other doco? the snippet in readme is kinda ALL you do to plug it in anyway

In most cases I see the explicit callout to autorest to be deprecated in favor of this approach, it's much cleaner and leaves the setup in Startup rather than in any of the consuming classes. But an example in the template would help clear that up.

**Comment by VaclavK** [comment]
> 
> 
> > > Looks good, just one small query on name of method. Would certainly be worth the effort to introduce an example of this into the template as well.
> > 
> > 
> > would that not duplicate autorest and other doco? the snippet in readme is kinda ALL you do to plug it in anyway
> 
> In most cases I see the explicit callout to autorest to be deprecated in favor of this approach, it's much cleaner and leaves the setup in Startup rather than in any of the consuming classes. But an example in the template would help clear that up.

I guess it boils down to how many APIs call other REST points... cause ultimately the template will not be able to "cover" all combinations/flavours/features and if it tried, it would move into example project rather than template

**Pull request merged**